### ISSUE TITLE
loki: query splitting, TODO

### ISSUE THREAD

**Issue by gabor** [grafana/grafana#63174]
here we list the issues that we still need to solve for this feature, missing polish, etc.

- [x] (FIXME: this may not be needed) consider scenarios where there are multiple loki queries:
    - [ ] with a mix of logs and metric queries
    - [ ] with a mix of metric queries with different resolution values
- [x] investigate if our frame-joining code is robust enough
    - [x] do we have good unit test coverage?
    - [ ] can we assume the field-order is always the same (check both metric and logs frames)
    - [x] can we assume all field-values can be just concatenated?
        - [x] check what happens with ID-generation for logs-fields, this may cause duplicate IDs.. or not.. need to check
- [x] make sure logs-volume queries work https://github.com/grafana/grafana/pull/63181
- [x] make sure logs-sample queries work
- [x] #63176
- [x] query-inspector panel:
    - `query` tab is not correct when query-splitting
    - `stats` panel only shows bytes-processed when quey-splitting
- [x] Testdata datasource, improve it's logs-mode to be able to do "streaming logs" (perhaps a checkbox in the query builder?), this is needed for the next step
- [x] logs-visualization panel, make sure it works correctly, some areas to check #63178
    - what happens if you open a log-line and click on a [+] in it, should the original streaming query be cancelled?
    - there are some calculations like “unique labels” that will have to be updated when new data arrives
    - does paging make sense in a streaming scenario?
    - is there any visual indication that the streaming is already happening?
    - you can alter the "direction" of the logs. what happens if you press the button and the data is still streaming
- [x] request cancelling. when the use presses the run-query button during the query happening, meaning the user stops the query, two things have to happen:
    - the currently running http-request has to be cancelled
    - no following http-requests should happen for the given query
- [x] when the code is complete&done, do a fresh read of the whole diff, and make sure the code quality is good:
    - variable names make sense
    - the location of functions makes sense (should something be moved to a new/different file? etc.)
    - we have enough unit tests

Part of https://github.com/grafana/grafana/issues/61768

**Comment by matyax** [comment]
@gabor not sure what you meant with:

> Testdata datasource, improve it's logs-mode to be able to do "streaming logs" (perhaps a checkbox in the query builder?), this is needed for the next step

**Comment by matyax** [comment]
@gabor should we close this one?

**Comment by gabor** [comment]
@matyax i'd recommend doing the last step on the list, which is basically reading the whole "diff", and making sure it makes sense, we use the same names everywhere, things are in the correct files, etc. for example:
- at the beginnings we used to say "query splitting" , i think later we went with "query chunking" , in that case `querySplitting.ts` should be called `queryChunking.ts` (or not?)
- `querySplitting.ts` has two functions `runGroupedCueries` and `runPartitionedQueries`... are these names ok? should one of them be called `runChunkedQueries`?
- we have a function named `requestSupportsPartitioning` .. should this be called `requestSupportsChunking` ?
- is the name `partitionTimeRange` good?
- etc.
